### ISSUE TITLE
About ApproxRepSet

### ISSUE THREAD

**Issue by hkmztrk** [giannisnik/repset#1]
Thanks a lot for sharing your code! I have a question about `ApproxRepSet` implementation:

Here you are computing `log_softmax`, 
https://github.com/giannisnik/repset/blob/208946423cb7a5037d42a962711d44c0c07ead2c/approxrepset/models.py#L31

which is then fed into `cross_entropy` function here,
https://github.com/giannisnik/repset/blob/208946423cb7a5037d42a962711d44c0c07ead2c/approxrepset/main.py#L58 

[According to torch](https://pytorch.org/docs/stable/generated/torch.nn.functional.cross_entropy.html#torch.nn.functional.cross_entropy), `cross_entropy` combines   `log_softmax` and `nll_loss`. So I was wondering whether double use of `log_softmax` was intentional? If so, would you mind explaining?

**Comment by giannisnik** [comment]
Hi @hkmztrk ,

Thank you for spotting this. Double use of `log_softmax` was actually not intentional. In fact, applying `log_softmax` twice produces the same output as applying it once. Therefore, you can safely replace `return F.log_softmax(out, dim=1)` with `return out`. This will have no impact on the output of the loss function.

**Comment by hkmztrk** [comment]
Thanks for fast  reply @giannisnik !
